ui-src/src/scoring.ts

```diff
@@ -7,44 +7,51 @@ export const CLICK_NON_MINE = 1;
 export const FLAG_MINE = 5;
 export const FLAG_NON_MINE = -2;
 
-export const getScore = (gameBoard: GameBoard, actions: Action[], agentHash: Hash) => {
-
-  const scoreReducer = (acc: number, action: Action): number => {
-    switch(action.actionType) {
-      case "flag":
-        if(isMine(gameBoard, action.position)) {
-          return acc + FLAG_MINE;
-        } else {
-          return acc + FLAG_NON_MINE;
-        }
-      case "reveal":
-        if(isMine(gameBoard, action.position)) {
-          return acc + CLICK_MINE;
-        } else {
-          return acc + CLICK_NON_MINE;
-        }
+// returns the scores for all players that have made moves in the game
+export const getScores = (gameBoard: GameBoard, actions: Action[]): Map<Hash, number> => {
+  type scoreMap = Map<Hash, number>
+
+  return actions.reduce<scoreMap>((scores: scoreMap, action: Action, index: number): scoreMap => {
+
+    let newScore = scores.get(action.agentHash);
+    if (newScore === undefined) { newScore = 0; }
+
+    if (isFirst(action, index, actions)) {
+      switch(action.actionType) {
+        case "flag":
+          if(isMine(gameBoard, action.position)) {
+            newScore += FLAG_MINE;
+          } else {
+            newScore += FLAG_NON_MINE;
+          }
+          break;
+        case "reveal":
+          if(isMine(gameBoard, action.position)) {
+            newScore += CLICK_MINE;
+          } else {
+            newScore += CLICK_NON_MINE;
+          }
+          break;
+        default:
+          break;
+      }
```

Comment:
Should we break this out into another method of CellMatrix like `actionScore = (action: Action, pos: Pos): number` so we can use this logic to calculate scores on the fly in the UI?